Production/govuk_ios/ViewModels/Topics/TopicsOnboardingWidgetViewModel.swift

```diff
@@ -0,0 +1,63 @@
+import Foundation
+import CoreData
+
+final class TopicsOnboardingWidgetViewModel {
+    private let analyticsService: AnalyticsServiceInterface
+    private let topicsService: TopicsServiceInterface
+    private let topicAction: ((Topic) -> Void)?
+    private let userDefaults: UserDefaults
+    private(set) var downloadError: TopicsServiceError?
+
+    var allTopics: [Topic] {
+        topicsService.fetchAllTopics()
+    }
+
+    init(topicsService: TopicsServiceInterface,
+         analyticsService: AnalyticsServiceInterface,
+         userDefaults: UserDefaults,
+         topicAction: ((Topic) -> Void)?) {
+        self.topicsService = topicsService
+        self.userDefaults = userDefaults
+        self.topicAction = topicAction
+        self.analyticsService = analyticsService
+        self.fetchTopics()
+    }
+
+    private func fetchTopics() {
+        topicsService.downloadTopicsList { result in
+            if case .failure(let error) = result {
+                self.downloadError = error
+            }
+        }
+    }
+
+    var getTopics: [Topic] {
```

Comment:
The get topics method has been moved to the topics widget viewModel, where it originally was , and will be used by the topic widget view, this will ensure that when the user skips, it will return all items

Reply:
the reason why the skip button caused no topics to be seen on the home page was because it was using the favourites lists to populate it